### PR TITLE
GH-15983: Craft RuleFit Examples [nocheck] [nochecks]

### PR DESCRIPTION
**Pull request by shaunyogeshwaran**
#15983 
Crafted new examples for the RuleFit algorithm to showcase the different parameters.

### CODE REVIEW

**Review by hannah-tillman — APPROVED** [review]

**Review by hannah-tillman — APPROVED** [review]

**Review by hannah-tillman — APPROVED** [review]

**Review by hannah-tillman — APPROVED** [review]

File: h2o-bindings/bin/custom/python/gen_rulefit.py [hunk 1]
```diff
@@ -52,3 +52,59 @@ def predict_rules(self, frame, rule_ids):
...
+
+examples = dict(
+
+    algorithm="""
+>>> import h2o
+>>> h2o.init()
+>>> from h2o.estimators import H2ORuleFitEstimator
+>>> f = "https://s3.amazonaws.com/h2o-public-test-data/smalldata/gbm_test/titanic.csv"
+>>> df = h2o.import_file(path=f, col_types={'pclass': "enum", 'survived': "enum"})
+>>> train, test = df.split_frame(ratios=[0.8], seed=1)
```
Comment (wendycwong) [inline h2o-bindings/bin/custom/python/gen_rulefit.py:64]: Remove line 64: >>> train, test = df.split_frame(ratios=[0.8], seed=1)

File: h2o-bindings/bin/custom/python/gen_rulefit.py [hunk 2]
```diff
@@ -52,3 +52,59 @@ def predict_rules(self, frame, rule_ids):
...
+>>> x = ["age", "sibsp", "parch", "fare", "sex", "pclass"]
+>>> y = "survived"
+>>> rfit = H2ORuleFitEstimator(max_rule_length=10,
+...                            max_num_rules=100,
+...                            algorithm="auto",
+...                            seed=1)
+>>> rfit.train(training_frame=train, x=x, y=y)
```
Comment (wendycwong) [inline h2o-bindings/bin/custom/python/gen_rulefit.py:71]: change
>>> rfit.train(training_frame=train, x=x, y=y)
to
>>> rfit.train(training_frame=df, x=x, y=y)

File: h2o-bindings/bin/custom/python/gen_rulefit.py [hunk 3]
```diff
@@ -52,3 +52,59 @@ def predict_rules(self, frame, rule_ids):
...
+>>> print(rfit.rule_importance())
+>>> rfit.predict(test)
```
Comment (wendycwong) [inline h2o-bindings/bin/custom/python/gen_rulefit.py:73]: remove line rfit.predict(test)

File: h2o-bindings/bin/custom/python/gen_rulefit.py [hunk 4]
```diff
@@ -52,3 +52,59 @@ def predict_rules(self, frame, rule_ids):
...
+
+""",
+    max_categorical_levels="""
+>>> import h2o
+>>> h2o.init()
+>>> from h2o.estimators import H2ORuleFitEstimator
+>>> f = "https://s3.amazonaws.com/h2o-public-test-data/smalldata/gbm_test/titanic.csv"
+>>> df = h2o.import_file(path=f, col_types={'pclass': "enum", 'survived': "enum"})
+>>> train, test = df.split_frame(ratios=[0.8], seed=1)
```
Comment (wendycwong) [inline h2o-bindings/bin/custom/python/gen_rulefit.py:82]: remove line: >>> train, test = df.split_frame(ratios=[0.8], seed=1)

File: h2o-bindings/bin/custom/python/gen_rulefit.py [hunk 5]
```diff
@@ -52,3 +52,59 @@ def predict_rules(self, frame, rule_ids):
...
+>>> x = ["age", "sibsp", "parch", "fare", "sex", "pclass"]
+>>> y = "survived"
+>>> rfit = H2ORuleFitEstimator(max_rule_length=10,
+...                            max_num_rules=100,
+...                            max_categorical_levels=10,
+...                            seed=1)
+>>> rfit.train(training_frame=train, x=x, y=y)
```
Comment (wendycwong) [inline h2o-bindings/bin/custom/python/gen_rulefit.py:89]: change

rfit.train(training_frame=train, x=x, y=y)

File: h2o-bindings/bin/custom/python/gen_rulefit.py [hunk 6]
```diff
@@ -52,3 +52,59 @@ def predict_rules(self, frame, rule_ids):
...
+>>> print(rfit.rule_importance())
+>>> rfit.predict(test)
```
Comment (wendycwong) [inline h2o-bindings/bin/custom/python/gen_rulefit.py:91]: remove >>> rfit.predict(test)

File: h2o-bindings/bin/custom/python/gen_rulefit.py [hunk 7]
```diff
@@ -52,3 +52,59 @@ def predict_rules(self, frame, rule_ids):
...
+""",
+    max_num_rules="""
+>>> import h2o
+>>> h2o.init()
+>>> from h2o.estimators import H2ORuleFitEstimator
+>>> f = "https://s3.amazonaws.com/h2o-public-test-data/smalldata/gbm_test/titanic.csv"
+>>> df = h2o.import_file(path=f, col_types={'pclass': "enum", 'survived': "enum"})
+>>> train, test = df.split_frame(ratios=[0.8], seed=1)
```
Comment (wendycwong) [inline h2o-bindings/bin/custom/python/gen_rulefit.py:99]: remove line with >>> train, test = df.split_frame(ratios=[0.8], seed=1)

File: h2o-bindings/bin/custom/python/gen_rulefit.py [hunk 8]
```diff
@@ -52,3 +52,59 @@ def predict_rules(self, frame, rule_ids):
...
+>>> x = ["age", "sibsp", "parch", "fare", "sex", "pclass"]
+>>> y = "survived"
+>>> rfit = H2ORuleFitEstimator(max_rule_length=10,
+...                            max_num_rules=100,
+...                            max_num_rules=-1,
+...                            seed=1)
+>>> rfit.train(training_frame=train, x=x, y=y)
```
Comment (wendycwong) [inline h2o-bindings/bin/custom/python/gen_rulefit.py:106]: change
>>> rfit.train(training_frame=train, x=x, y=y)
to
>>> rfit.train(training_frame=df, x=x, y=y)

File: h2o-bindings/bin/custom/python/gen_rulefit.py [hunk 9]
```diff
@@ -52,3 +52,59 @@ def predict_rules(self, frame, rule_ids):
...
+>>> print(rfit.rule_importance())
+>>> rfit.predict(test)
```
Comment (wendycwong) [inline h2o-bindings/bin/custom/python/gen_rulefit.py:108]: remove >>> rfit.predict(test)

**Review by wendycwong — CHANGES_REQUESTED** [review]

**Review by valenad1 — CHANGES_REQUESTED** [review]
I agree with wendy, I would also use different values than the default in the examples.

**Review by maurever — CHANGES_REQUESTED** [review]
Why did you add examples only to the `max_categorical_levels`,` algorithm`, and `max_num_rules` parameters?

It would be nice to add examples also for:
- min_rule_lenght
- max_rule_length
- model_type
- distribution
- rule_generation_ntrees
- rule_importance
- predict_rules

**Comment by wendycwong** [comment]
Good call Veronika, I missed those rulefit specific parameters.

**Comment by shaunyogeshwaran** [comment]
@maurever What would some values for `rule_importance` and ` predict_rules` be? The docs don't give much info

**Comment by valenad1** [comment]
> @maurever What would some values for `rule_importance` and ` predict_rules` be? The docs don't give much info

`rule_importance` is just output. So train the model and call `model.rule_importance` and same for `predict_rules`. E.g [here](https://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-science/rulefit.html?highlight=rulefit#examples).

**Comment by maurever** [comment]
Thanks, @valenad1, you are correct. It is my mistake to conclude them in the list of parameters. If I am correct, we can also add examples of how to use these functions. WDYT, @shaunyogeshwaran ?

**Comment by shaunyogeshwaran** [comment]
> If I am correct, we can also add examples of how to use these functions. WDYT, @shaunyogeshwaran ?

Agreed, that is something I will be able to look into. @hannah-tillman what are your thoughts on this?

**Comment by hannah-tillman** [comment]
> Agreed, that is something I will be able to look into. @hannah-tillman what are your thoughts on this?

@shaunyogeshwaran Yeah adding those examples in won't be an issue. Both of those calls are in written out in the gen_rulefit.py page so instead of adding them to the bottom of the examples list you'll just add them to their respective sections like this under ``:return:``

```
:examples:
>>> example content
>>> example content
>>> etc
"""
```

File: h2o-bindings/bin/custom/python/gen_rulefit.py [hunk 10]
```diff
@@ -52,3 +52,157 @@ def predict_rules(self, frame, rule_ids):
...
+    max_num_rules="""
+>>> import h2o
+>>> h2o.init()
+>>> from h2o.estimators import H2ORuleFitEstimator
+>>> f = "https://s3.amazonaws.com/h2o-public-test-data/smalldata/gbm_test/titanic.csv"
+>>> df = h2o.import_file(path=f, col_types={'pclass': "enum", 'survived': "enum"})
+>>> x = ["age", "sibsp", "parch", "fare", "sex", "pclass"]
+>>> y = "survived"
+>>> rfit = H2ORuleFitEstimator(max_rule_length=10,
+...                            max_num_rules=-2,
```
Comment (wendycwong) [inline h2o-bindings/bin/custom/python/gen_rulefit.py:98]: please set

max_num_rules = 3 

or some positive number

File: h2o-bindings/bin/custom/python/gen_rulefit.py [hunk 11]
```diff
@@ -52,3 +52,157 @@ def predict_rules(self, frame, rule_ids):
...
+>>> x = ["age", "sibsp", "parch", "fare", "sex", "pclass"]
+>>> y = "survived"
+>>> rfit = H2ORuleFitEstimator(max_rule_length=10,
+...                            max_num_rules=100,
+...                            rule_generation_ntrees=60,
+...                            seed=1)
+>>> rfit.train(training_frame=df, x=x, y=y)
+>>> print(rfit.rule_importance())
+""",
+    predict_rules="""
```
Comment (wendycwong) [inline h2o-bindings/bin/custom/python/gen_rulefit.py:193]: Please add how you will use 

predict_rules.

**Review by wendycwong — CHANGES_REQUESTED** [review]

**Review by wendycwong — CHANGES_REQUESTED** [review]

**Review by hannah-tillman — DISMISSED** [review]

**Comment by wendycwong** [comment]
Jenkins still cannot run.  Please change github branch name and remove the #.

**Pull request closed without merging**